File: src/cpp/explore/topquestions/easy/array/删除排序数组中的重复项.cpp
/**
 * @File    :   删除排序数组中的重复项.cpp
 * @Time    :   2020/07/24 07:30:36
 * @Author  :   wylu
 * @Version :   1.0
 * @Contact :   15wylu@gmail.com
 * @License :   Copyright © 2020, wylu-CHINA-SHENZHEN. All rights reserved.
 * @Desc    :   https://leetcode-cn.com/explore/featured/card/top-interview-questions-easy/1/array/21/
 */

#include <vector>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) {
            return 0;
        }

        int ans = 1;
        for (int i = 1, j = 0; i < nums.size(); i++) {
            if (nums[i] != nums[j]) {
                nums[++j] = nums[i];
                ans++;
            }
        }
        return ans;
    }
};

File: src/python/explore/topquestions/easy/array/删除排序数组中的重复项.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
@File    :   删除排序数组中的重复项.py
@Time    :   2020/07/26 09:16:10
@Author  :   wylu
@Version :   1.0
@Contact :   15wylu@gmail.com
@License :   Copyright © 2020, wylu-CHINA-SHENZHEN. All rights reserved.
@Desc    :
"""
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0

        i, j = 0, 1
        while j < len(nums):
            if nums[j] != nums[i]:
                i += 1
                nums[i] = nums[j]
            j += 1
        return i + 1
